source/contributed/typescript-and-screeps-1.md

```diff
@@ -0,0 +1,207 @@
+title: TypeScript and Screeps, from Beginner to Advanced, Pt. 1
+contributed:
+    name: bryan
+    link: https://github.com/bryanbecker
+    date: 2017-05-15
+---
+
+# TypeScript and Screeps, from Beginner to Advanced, Pt. 1
+
+* [Part 1](http://docs.screeps.com/contributed/typescript-and-screeps-1.html)
+**Introduction to TypeScript and setting up a TypeScript + Screeps environment**
+* [Part 2](http://docs.screeps.com/contributed/typescript-and-screeps-2.html)
+In depth look using TypeScript's features to extend the built-in Screeps objects and methods
+* [Part 3](http://docs.screeps.com/contributed/typescript-and-screeps-3.html)
+Getters, setters, and meta programming with decorators
+
+Part One of this series with introduce setting up a TypeScript project for Screeps.  This article will present a basic guide to setting up a TypeScript project for Screeps, and a brief introduction to third party tools.
+
+For users who already have a working TypeScript setup, you can directly skip to [Part 2](http://docs.screeps.com/contributed/typescript-and-screeps-2.html).
+
+For a quick-start, skip down to [screeps-typescript-starter](#screeps-typescript-starter).
+
+## Why TypeScript?
+
+TypeScript is a superset of ECMAScript (a fancy way to say Javascript).  That means it has *all* the features of ECMA, *plus* some extra ones.  When a project is compiled, Typescript will process all your files, and "transpile" them into the target Javascript version (ES6, in the case of Screeps).  This works similarly to Babel, if you have used that in the past.
+
+Using TypeScript can be frustrating when you first start, because it seems the compiler never stops complaining. Power through it. This ends up being the most useful feature of TypeScript once you get going. Towards then, we'll look at some concrete examples of this.
+
+This guide won't attempt to teach you typescript ([Typescript Deep Dive](https://basarat.gitbooks.io/typescript/docs/why-typescript.html) is an execellent starting resource for that), but will provide an introduction in getting set up, and using the more advanced features to your advantage
+
+## Getting Started with TypeScript
+
+> **Note**: If you already have a Typescript project that compiles, and understand the basic tooling, you can most likely skip the rest of this article
+
+This section will only provide a very brief overview of getting started with TypeScript.  For more in-depth, please see the above Deep Dive link, or the TypeScript docs.
+
+### Install Your Tooling
+
+To get off the ground with typescript, you're going to need quite a few different tools:
+
+* An IDE that supports TS ([Slant](https://www.slant.co/topics/5815/~ides-for-typescript-development) recommends Visual Studio Code as the most popular choice)
+
+* NPM (not in the scope of this article)
+
+* The typescript compiler:
+  ```bash
+  # for cutting edge in development version (currently 2.4.0):
+  $ npm install -g typescript@next
```

Comment:
I personally really dislike global installations as they put a constraint on the host system.

I do realise though that it can be a lot more tedious to install everything locally (especially without covering npm), so this might be okay in this case.
Alternatively you could just provide the links to the tools and let the use install them himself.